### PR TITLE
Studio: Display spinner when the site is being pulled

### PR DESCRIPTION
**Pull request by ivan-ottinger**
## Related issues

- Fixes https://github.com/Automattic/dotcom-forge/issues/9897.

## Proposed Changes

- display spinner when the site is being pulled, use `Syncing` tooltip
- add tooltip messages for other cases where the spinner is displayed:
  - `Adding` - when a new site is being added
  - `Importing` - when an exported site is being imported
  - `Loading` - fallback tooltip message 

Markup on 2024-11-19 at 17:37:43

## Testing Instructions

1. Check out the PR and build the app with `STUDIO_SITE_SYNC=true npm start`.
2. Open the Sync tab.
3. Connect a WP.com site to Studio.
4. Pull the site from WP.com.
5. Observe the spinner by the site in the sidebar. 

Testing other tooltip messages:

1. Go through the process of adding a new site and observe the `Adding` tooltip when the site is being added.
2. Import an exported site and observe the `Importing` tooltip as the site is being imported.

## Pre-merge Checklist

- [x] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

**Review by katinthehatsite — APPROVED** [review]
The functionality looks good to me and works as expected 👍 
Nice work!

I noticed that some unit tests failed and tried re-retriggering them. They might pass but otherwise, would need to be adjusted before merging.

**Comment by ivan-ottinger** [comment]
Thank you for your review and testing, Kat.

> I noticed that some unit tests failed and tried re-retriggering them. They might pass but otherwise, would need to be adjusted before merging.

Oh, yeah, I missed running unit tests before asking for review yesterday evening. Sorry about that. I have now fixed the unit tests in https://github.com/Automattic/studio/pull/672/commits/123890cb6fb40996971e52d8383c6b99251e8517 (and subsequent lint fix https://github.com/Automattic/studio/pull/672/commits/5f328958bf7cb84d4bd25bf4f1dd050af4e6a08f).

We are also waiting for feedback from Matt and Lucas now (https://github.com/Automattic/dotcom-forge/issues/9897#issuecomment-2487994378), so more changes may come in.

I will share another comment here once the PR is ready for another review round.

**Comment by ivan-ottinger** [comment]
This PR is ready for another review round.

File: src/components/site-menu.tsx
Comment (wojtekn) [inline src/components/site-menu.tsx]: Could we avoid using nested ternary and use if-else to make it more readable?

**Review by wojtekn — APPROVED** [review]
Works well, and the code change looks good. I left one note regarding code readability.

File: src/components/site-menu.tsx
Comment (ivan-ottinger) [reply src/components/site-menu.tsx]: Sounds good. I have made the change in https://github.com/Automattic/studio/pull/672/commits/a35255914338a3a67fb17022c32f6031c82d4df1.

**Pull request merged**
